File: src/ui_ng/src/app/repository/repository.ts
// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
/*
  {
    "id": "2",
    "name": "library/mysql",
    "owner_id": 1,
    "project_id": 1,
    "description": "",
    "pull_count": 0,
    "star_count": 0,
    "tags_count": 1,
    "creation_time": "2017-02-14T09:22:58Z",
    "update_time": "0001-01-01T00:00:00Z"
  }
*/

export class Repository {
  id: number;
  name: string;
  owner_id: number;
  project_id: number;
  description: string;
  pull_count: number;
  start_count: number;
  tags_count: number;
  creation_time: Date;
  update_time: Date;

  constructor(name: string, tags_count: number) {
    this.name = name;
    this.tags_count = tags_count;
  }
}

File: src/server/v2.0/handler/model/repository.go
// Copyright Project Harbor Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package model

import (
	"github.com/go-openapi/strfmt"

	"github.com/goharbor/harbor/src/pkg/repository/model"
	"github.com/goharbor/harbor/src/server/v2.0/models"
)

// RepoRecord model
type RepoRecord struct {
	*model.RepoRecord
}

// ToSwagger converts the repository into the swagger model
func (r *RepoRecord) ToSwagger() *models.Repository {
	var createTime *strfmt.DateTime
	if !r.CreationTime.IsZero() {
		t := strfmt.DateTime(r.CreationTime)
		createTime = &t
	}

	return &models.Repository{
		CreationTime: createTime,
		Description:  r.Description,
		ID:           r.RepositoryID,
		Name:         r.Name,
		ProjectID:    r.ProjectID,
		PullCount:    r.PullCount,
		UpdateTime:   strfmt.DateTime(r.UpdateTime),
	}
}

// NewRepoRecord ...
func NewRepoRecord(r *model.RepoRecord) *RepoRecord {
	return &RepoRecord{RepoRecord: r}
}
